internal/cmd/controller/reconciler/bundle_controller.go

```diff
@@ -150,6 +189,11 @@ func (r *BundleReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		// NOTE we don't use the existing BundleDeployment, we discard annotations, status, etc
 		// copy labels from Bundle as they might have changed
 		bd := target.BundleDeployment()
+
+		if bd.DeletionTimestamp.IsZero() {
+			controllerutil.AddFinalizer(bd, bundleDeploymentFinalizer)
+		}
```

Comment:
In what case is this going to be different to zero?
In my understanding the `BundleDeployment` struct was just instantiated and it's still not added to the cluster.

I guess that we could omit the `if` statement if that's correct

Reply:
You are right, that condition was here as a precaution; I have replaced it with a comment explaining precisely that.